From a0c886bca0cc9fa325384ca40cbd51a44054284d Mon Sep 17 00:00:00 2001
From: Vincent Tinelli <vincent.tinelli@intel.com>
Date: Thu, 9 Aug 2012 12:21:53 +0200
Subject: [PATCH 13/49] [PORT FROM MAIN] Expand variables on mount_all command

BZ: 50823

Reason:
Generic mount_all command in init.common.rc
instead of duplicating in init.${ro.hardware}
fstab.${ro.hardware} needed for vold/cryptfs

Implementation:
Expand variables on mount_all command

Change-Id: I15f2443d2aae71ddfc8d2e24fc84584662b498ec
Signed-off-by: Vincent Tinelli <vincent.tinelli@intel.com>
Reviewed-on: http://android.intel.com:8080/76542
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Romieu, Benoit <benoit.romieu@intel.com>
Tested-by: Romieu, Benoit <benoit.romieu@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 init/builtins.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/init/builtins.c b/init/builtins.c
index aaf85d9..aff479a 100644
--- a/init/builtins.c
+++ b/init/builtins.c
@@ -466,6 +466,8 @@ int do_mount_all(int nargs, char **args)
     int child_ret = -1;
     int status;
     const char *prop;
+    char prop_val[PROP_VALUE_MAX];
+
 
     if (nargs != 2) {
         return -1;
@@ -489,7 +491,12 @@ int do_mount_all(int nargs, char **args)
     } else if (pid == 0) {
         /* child, call fs_mgr_mount_all() */
         klog_set_level(6);  /* So we can see what fs_mgr_mount_all() does */
-        child_ret = fs_mgr_mount_all(args[1]);
+        ret = expand_props(prop_val, args[1], sizeof(prop_val));
+        if (ret) {
+            ERROR("cannot expand '%s' while assigning to '%s'\n", args[1], prop_val);
+            return -1;
+        }
+        child_ret = fs_mgr_mount_all(prop_val);
         if (child_ret == -1) {
             ERROR("fs_mgr_mount_all returned an error\n");
         }
-- 
1.7.9

